input/kafkamdm/lag_monitor.go

```diff
@@ -47,43 +48,107 @@ func (l *lagLogger) Min() int {
 	return min
 }
 
+type rateLogger struct {
+	sync.Mutex
+	lastOffset int64
+	lastTs     time.Time
+	rate       int64
+}
+
+func (o *rateLogger) Store(offset int64, ts time.Time) {
+	o.Lock()
+	defer o.Unlock()
+	if o.lastTs.IsZero() {
+		// first measurement
+		o.lastOffset = offset
+		o.lastTs = ts
+		return
+	}
+	metrics := offset - o.lastOffset
+
+	o.lastOffset = offset
+
+	duration := int64(ts.Sub(o.lastTs) / time.Second)
+	o.lastTs = ts
+	if duration <= 0 {
+		// current ts is > last ts. This would only happen if our clock
+		// suddenly changes, in which case we cant reliably work out how
+		// low it has really been since we last took a measurement.
+		return
+	}
+	rate := metrics / duration
```

Comment:
if we kept duration in milliseconds until here, and then rounded after this step, our results could be up to 20% more accurate (assuming we do this every 5 seconds)